presto-hive/src/main/java/com/facebook/presto/hive/HiveColumnHandle.java

```diff
@@ -225,6 +236,11 @@ public static HiveColumnHandle fileModifiedTimeColumnHandle()
         return new HiveColumnHandle(FILE_MODIFIED_TIME_COLUMN_NAME, FILE_MODIFIED_TIME_TYPE, FILE_MODIFIED_TIME_TYPE_SIGNATURE, FILE_MODIFIED_TIME_COLUMN_INDEX, SYNTHESIZED, Optional.empty(), ImmutableList.of(), Optional.empty());
     }
 
+    public static boolean isRowIdColumnHandle(HiveColumnHandle column)
+    {
+        return column.getHiveColumnIndex() == ROW_ID_COLUMN_INDEX && "$row_id".equals(column.getName());
```

Comment:
did you ever figure out why you were seeing -16 as the column index for a regular column?

Reply:
No, I have not but it's definitely a thing, at least in tests.

Reply:
Gah, okay this shouldn't happen so i dug into it and it's a real bug!  partition key columns use ids starting from -13 and more negative (see https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/HiveColumnHandle.java#L67-L68, and https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java#L987-L993).  Hidden columns  should be using values -1  through -12.  And your change (and the -14 and -15 indexed special columns) are using values that can be shared with partition columns.  I think the -14 and -15 column indexes haven't run into this issue because they are using the "prefilled" path, which checks first if something is a partition key and then checks special columns.